docs/caching.md

```diff
@@ -1,43 +1,11 @@
 ## Caching
 
-Relay is responsible for graphql response caching, which happens as users navigate from page to page on the client. This is typically configured via the route.
```

Comment:
These are not actual changes! 

Noticed the old caching doc was around (related to v1 redis proxy approach), so deleted that file and renamed caching2 (that matt worked on) to cache.md.